include/asterisk/app.h

```diff
@@ -123,6 +123,8 @@ enum ast_timelen {
 */
 int ast_ivr_menu_run(struct ast_channel *c, struct ast_ivr_menu *menu, void *cbdata);
 
+char *ast_app_filename_list_iterate(char **restrict stringp);
```

Comment:
Documentation would be nice...
Also XML documentation for escaping to the dialplan applications (at least Playback), like you commented on the issue.